Dockerfile

```diff
@@ -1,119 +1,102 @@
-
-RUN git clone https://github.com/genouest/biomaj-user.git
-
-RUN git clone https://github.com/genouest/biomaj-cli.git
-
-RUN git clone https://github.com/genouest/biomaj-process.git
-
-RUN git clone https://github.com/genouest/biomaj-download.git
-
-RUN git clone  https://github.com/genouest/biomaj.git && echo "Install biomaj"
-
-RUN git clone https://github.com/genouest/biomaj-daemon.git && echo "Install daemon"
-
-RUN git clone https://github.com/genouest/biomaj-watcher.git && echo "Install biomaj-watcher"
-
-RUN git clone https://github.com/genouest/biomaj-ftp.git
-
-RUN git clone https://github.com/genouest/biomaj-release.git
-
-RUN git clone https://github.com/genouest/biomaj-data.git
+RUN git clone https://github.com/genouest/biomaj-core.git && \
+    git clone https://github.com/genouest/biomaj-zipkin.git && \
+    git clone https://github.com/genouest/biomaj-user.git && \
+    git clone https://github.com/genouest/biomaj-cli.git && \
+    git clone https://github.com/genouest/biomaj-process.git && \
+    git clone https://github.com/genouest/biomaj-download.git && \
+    git clone https://github.com/genouest/biomaj.git && \
+    git clone https://github.com/genouest/biomaj-daemon.git && \
+    git clone https://github.com/genouest/biomaj-watcher.git && \
+    git clone https://github.com/genouest/biomaj-ftp.git && \
+    git clone https://github.com/genouest/biomaj-release.git && \
+    git clone https://github.com/genouest/biomaj-data.git
 
 ENV BIOMAJ_CONFIG=/etc/biomaj/config.yml
 
 RUN mkdir -p /var/log/biomaj
 
-RUN pip3 install greenlet==0.4.17
-RUN pip3 install gevent==1.4.0
-RUN pip3 install graypy
-RUN pip3 install pymongo==3.12.3
-RUN pip3 install redis==3.5.3
-RUN pip3 install wheel
-RUN pip3 install PyYAML==5.3.1
+RUN pip3 install --no-cache-dir pip --upgrade && \
+    pip3 install --no-cache-dir setuptools --upgrade && \
+    pip3 install --no-cache-dir greenlet==0.4.17 && \
+    pip3 install --no-cache-dir gevent==1.4.0 && \
+    pip3 install --no-cache-dir graypy && \
+    pip3 install --no-cache-dir pymongo==3.12.3 && \
+    pip3 install --no-cache-dir redis==3.5.3 && \
+    pip3 install --no-cache-dir wheel && \
+    pip3 install --no-cache-dir PyYAML==5.4.1 && \
+    pip3 install --no-cache-dir protobuf==3.20.3 && \
+    python3 -m pip install --no-cache-dir ftputil
 
 ENV SUDO_FORCE_REMOVE=yes
-RUN buildDeps='gcc python3-dev protobuf-compiler' \
+RUN buildDeps="gcc python3-dev protobuf-compiler" \
     && set -x \
+    && apt-get update \
     && apt-get install -y $buildDeps --no-install-recommends \
-    && cd /root/biomaj-core && python3 setup.py build && pip3 install . \
-    && cd /root/biomaj-zipkin && python3 setup.py build && pip3 install . \
-    && cd /root/biomaj-user && python3 setup.py build && pip3 install . \
-    && cd /root/biomaj-cli && python3 setup.py build && pip3 install . \
+    && cd /root/biomaj-core && python3 setup.py build && pip3 install --no-cache-dir . \
```

Comment:
We should use this quicker way: `pip install git+https://github.com/genouest/biomaj-core.git` (for all of these modules)

Reply:
Nice.
Using this neat little trick, does pip even use the `setup.py` file from the repo that it builds its wheel from or does it look at the `requirements.txt` all by itself ?